File: solutions/ruby/clock/1/clock.rb
class Clock
  def initialize(hour: 0, minute: 0)
    @minutes = (hour * 60 + minute) % (24 * 60)
  end

  def to_s
    "%02d:%02d" % [@minutes / 60, @minutes % 60]
  end

  def +(other)
    Clock.new(minute: @minutes + other.minutes)
  end

  def -(other)
    Clock.new(minute: @minutes - other.minutes)
  end

  def ==(other)
    self.to_s == other.to_s
  end

  protected
  attr_reader :minutes
end

File: solutions/python/clock/2/clock.py
class Clock:
    def __init__(self, hour, minute):
        total = (hour * 60 + minute) % 1440
        self.h = total // 60
        self.m = total % 60

    def __repr__(self):
        return f"Clock({self.h}, {self.m})"

    def __eq__(self, other):
        return self.h == other.h and self.m == other.m

    def __add__(self, minutes):
        total = (self.h * 60 + self.m + minutes) % 1440
        return Clock(total // 60, total % 60)

    def __sub__(self, minutes):
        return self + (-minutes)
